src/node/txdownloadman_impl.cpp

```diff
@@ -174,9 +174,37 @@ void TxDownloadManagerImpl::DisconnectedPeer(NodeId nodeid)
 
 bool TxDownloadManagerImpl::AddTxAnnouncement(NodeId peer, const GenTxid& gtxid, std::chrono::microseconds now, bool p2p_inv)
 {
+    // If this is an orphan we are trying to resolve, consider this peer as a orphan resolution candidate instead.
+    // - received as an p2p inv
+    // - is wtxid matching something in orphanage
+    // - exists in orphanage
+    // - peer can be an orphan resolution candidate
+    if (p2p_inv && gtxid.IsWtxid()) {
+        if (auto orphan_tx{m_orphanage.GetTx(Wtxid::FromUint256(gtxid.GetHash()))}) {
```

Comment:
```Suggestion
        const auto wtxid{Wtxid::FromUint256(gtxid.GetHash()))};
        if (auto orphan_tx{m_orphanage.GetTx(wtxid)}) {
```
and use `wtxid` below in the other two locations

Reply:
Added in #31666